### PR TITLE
fix(UI): unset few field while create a duplicate in project and component tab

### PR DESCRIPTION
**Pull request by nikkuma7**
…onent

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? #1755 
> * Did you add or update any new dependencies that are required for your change?

Issue: 
When we do duplication in product and component (release) tab, few unnecessary information is also copied, so we unset those field as per the user request.

### How To Test?
1: Login sw360 
2: Go to project tab and create a duplicate.
3: Go to component -> release tab and create duplicate.

> How should these changes be tested by the reviewer?
> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Testing was successful.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java [hunk 1]
```diff
@@ -1080,6 +1080,9 @@ private void prepareReleaseDuplicate(RenderRequest request, RenderResponse respo
             Release release = PortletUtils.cloneRelease(emailFromRequest, client.getAccessibleReleaseById(releaseId, user));
             Map<String, String> sortedAdditionalData = getSortedMap(release.getAdditionalData(), true);
             release.setAdditionalData(sortedAdditionalData);
+            release.unsetModifiedBy();
+            release.unsetModifiedOn();
+            release.unsetAdditionalData();
```
Comment (akapti) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java:1085]: This isn't the right place, Please unset all the attributes in `cloneProject` method of  `PortletUtils` class.
screenshot for reference:
image

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 2]
```diff
@@ -2328,10 +2328,10 @@ private void prepareProjectDuplicate(RenderRequest request) {
                 String department = user.getDepartment();
                 Project newProject = PortletUtils.cloneProject(emailFromRequest, department, client.getProjectById(id, user));
                 setDefaultRequestAttributes(request, newProject.getBusinessUnit());
-                Map<String, String> sortedAdditionalData = getSortedMap(newProject.getAdditionalData(), true);
-                newProject.setAdditionalData(sortedAdditionalData);
                 setAttachmentsInRequest(request, newProject);
                 PortletUtils.setCustomFieldsEdit(request, user, newProject);
+                newProject.unsetModifiedBy();
+                newProject.unsetModifiedOn();
```
Comment (akapti) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:2334]: This isn't the right place, Please unset all the attributes in `cloneRelease` method of  `PortletUtils` class.

image

**Review by akapti — CHANGES_REQUESTED** [review]
Please incorporate the review comments.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java (hunk 1 above)
Comment (nikkuma7) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java]: done.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java (hunk 2 above)
Comment (nikkuma7) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java]: done.

**Review by akapti — CHANGES_REQUESTED** [review]
code looks good to me.
tested the changes, and working fine for duplication of release.

but duplication of projects with linked obligations is failing, `modified on` and `modified by` is present when duplicating a project with obligations.

**Review by akapti — APPROVED** [review]
Code looks good. Tested the changes, working as expected.

**Pull request merged**